Sources/Request/Request/RequestParams/CombinedParams.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  File.swift
+//  
+//
+//  Created by brennobemoura on 15/11/20.
+//
+
+import Foundation
+
+internal struct CombinedParams: RequestParam {
+    fileprivate let children: [RequestParam]
+
+    init(children: [RequestParam]) {
+        self.children = children
+    }
+
+    func buildParam(_ request: inout URLRequest) {
+        children.forEach {
+            $0.buildParam(&request)
+        }
+    }
```

Comment:
You should be able to remove unzip if you replace that with:
```suggestion
    func buildParam(_ request: inout URLRequest) {
        children
            .sorted { a, _ in (a is Url) }
            .filter { !($0 is SessionParam) }
            .forEach {
                $0.buildParam(&request)
            }
    }
    
    func buildConfiguration(_ configuration: URLSessionConfiguration) {
        children
            .compactMap { $0 as? SessionParam }
            .forEach {
                $0.buildConfiguration(configuration)
            }
    }
```
(and conform to `SessionParam`)

Reply:
I guess we can remove the fatal error on SessionParam since CombinedParams can be it, so it won't be filtered on buildParam.

Reply:
Maybe leave the fatalError in case of some unseen edge case?

Reply:
Hmmm, I will try using the Never swift type and filter it.

Reply:
Forget using Never, I add one more verification on filter:

```
children
    .sorted { a, _ in (a is Url) }
    .filter { !($0 is SessionParam) || $0 is CombinedParams }
    .forEach {
        $0.buildParam(&request)
    }
```